piccolo/utils/pydantic.py

```diff
@@ -90,6 +90,7 @@ def create_pydantic_model(
     deserialize_json: bool = False,
     recursion_depth: int = 0,
     max_recursion_depth: int = 5,
+    pydantic_extra_fields: str = "ignore",
```

Comment:
Sorry - one minor thing. Can we make this:

```python
pydantic_extra_fields: t.Literal['ignore', 'allow', 'forbid'] = "ignore",
```

Reply:
Originally I actually wanted to do this:

```
pydantic_extra_fields: pydantic.Extra = pydantic.Extra.ignore
```

Is there a stylistic or other reason for your suggested change? (I'm really curious, not trying to be confrontational)

Reply:
Yeah, that works too:

```python
pydantic_extra_fields: pydantic.Extra = pydantic.Extra.ignore
```

I don't think there's clearly a best. The reason I suggested `Literal` is it means a linter can detect typos in the string.

I think the `Enum` is probably the best though. It's slightly more verbose to use, but means that if Pydantic adds another option in the future, we don't have to update `Literal` to include the new options.

Ideally it would be good to expose all of Pydantic's config options. We could take a different approach entirely, and do this:

```python
def create_pydantic_model(pydantic_config: t.Type[BaseConfig] = BaseConfig, ...):
    ...
    class CustomConfig(pydantic_config):
        schema_extra = {
            "help_text": table._meta.help_text,
            **schema_extra_kwargs,
        }
        extra = pydantic.Extra(pydantic_extra_fields)
```

Which means that every Pydantic option is exposed to the user. What do you think?

Reply:
Looks good to me . I'll work on this.

Reply:
I see there are quite a few places in piccolo-api that import `Config` from `utils/pydantic`, IIUC that class would now go away since we'd now be instantiating just `CustomConfig` based on `pydantic_config`, shall I (try to) adjust those other places too?

Reply:
Ah, yeah - good point. I wonder if we keep `Config`, then this is possible?

```python
def create_pydantic_model(pydantic_config: t.Type[BaseConfig] = BaseConfig, ...):
    ...
    class CustomConfig(Config, pydantic_config):
        schema_extra = {
            "help_text": table._meta.help_text,
            **schema_extra_kwargs,
        }
        extra = pydantic.Extra(pydantic_extra_fields)
```

I think the `mro` still works, even though both classes inherit from `BaseConfig`.

Reply:
Sorry for being a PITA, but now I wonder what we should do with the original `pydantic_extra_fields` parameter, as now the user could just do

```
class MyConfig(pydantic.BaseConfig):
    extra = 'forbid'

model = create_pydantic_model(pydantic_config=MyConfig, ...)
```

and obtain the same result as passing `pydantic_extra_fields='forbid'`. I don't have a strong opinion on this, although using `pydantic_extra_fields` seems slightly more convenient. Comments?

Reply:
We probably don't need them both. I agree that having a `pydantic_extra_fields` argument is more convenient. But over time, as we support more Pydantic options, `create_pydantic_model` could end up with loads of arguments.

We could accept a dict, and anything passed into it automatically applied to the config (rather than having to pass in a config object):

```python
create_pydantic_model(pydantic_config={'extra': Extra.ignore})
```

It's more convenient, but there's no type checking. We could use `TypedDict`, but would constantly be updating it as Pydantic adds more arguments.

Sorry, I'm kind of thinking out loud here ... what do you think?